kvfollowerreadsccl: fix race in TestBoundedStalenessDataDriven

otan opened the pull request:
onStmtTrace can be racy as it may read `stmt` whilst we are setting the
`stmt` function in the test.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/69075)
<!-- Reviewable:end -->

otan committed:
kvfollowerreadsccl: fix race in TestBoundedStalenessDataDriven

onStmtTrace can be racy as it may read `stmt` whilst we are setting the
`stmt` function in the test.

Release note: None

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/69075#-:-MhMRsOTBjr5zV8JMxFY:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @arulajmani)



<!-- Sent from Reviewable.io -->

otan commented:
thanks for the quick review!

bors r=nvanbenschoten
